app/src/CoreGameLogic/Feature/Spielzug/Aktion/RepayLoanForPlayerAktion.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Domain\CoreGameLogic\Feature\Spielzug\Aktion;
+
+use Domain\CoreGameLogic\EventStore\GameEvents;
+use Domain\CoreGameLogic\EventStore\GameEventsToPersist;
+use Domain\CoreGameLogic\Feature\Moneysheet\State\LoanCalculator;
+use Domain\CoreGameLogic\Feature\Moneysheet\State\MoneySheetState;
+use Domain\CoreGameLogic\Feature\Moneysheet\ValueObject\LoanId;
+use Domain\CoreGameLogic\Feature\Spielzug\Aktion\Validator\HasPlayerEnoughResourcesValidator;
+use Domain\CoreGameLogic\Feature\Spielzug\Aktion\Validator\HasPlayerNotAlreadyRepaidLoanValidator;
+use Domain\CoreGameLogic\Feature\Spielzug\Dto\AktionValidationResult;
+use Domain\CoreGameLogic\Feature\Spielzug\Event\LoanWasRepaidForPlayer;
+use Domain\CoreGameLogic\PlayerId;
+use Domain\Definitions\Card\Dto\ResourceChanges;
+use Domain\Definitions\Card\ValueObject\MoneyAmount;
+use RuntimeException;
+
+class RepayLoanForPlayerAktion extends Aktion
+{
+    public function __construct(
+        private readonly LoanId $loanId,
+    ) {
+    }
+
+    public function validate(PlayerId $playerId, GameEvents $gameEvents): AktionValidationResult
+    {
+        $resourceChanges = $this->getResourceChanges($playerId, $gameEvents, $this->loanId);
```

Comment:
Wenn wir es gleich im Objekt speichern, müssen wir die Funktion nicht mehrfach durchrattern:
```suggestion
        $this->resourceChanges = $this->getResourceChanges($playerId, $gameEvents, $this->loanId);
```

Und in der `execute` benutzen wir dann gleich `$this->resourceChanges`

Reply:
validate und excecute können aber getrennt von außen aufgerufen werden. Mann müsste es im constructor rein schreiben aber da fehlen playerId und gameEvents. Außer ich sehs grad nicht.

Reply:
es reicht, wenn es in validate steht, weil validate immer von execute aufgerufen wird.

Reply:
Hm find ich irgendwie nicht geiler.